classes/table/adhoc_tasks.php

```diff
@@ -0,0 +1,147 @@
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Proxy lock factory, adhoc task summary table.
+ *
+ * @package    tool_lockstats
+ * @author     Ilya Tregubov <ilyatregubov@catalyst-au.net>
+ * @copyright  2019 Catalyst IT
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+namespace tool_lockstats\table;
+
+if (!defined('MOODLE_INTERNAL')) {
+    die('Direct access to this script is forbidden.'); // It must be included from a Moodle page.
+}
+
+use core_component;
+use html_table;
+use html_table_cell;
+use html_table_row;
+use html_writer;
+use moodle_url;
+use stdClass;
+
+/**
+ * Proxy lock factory, adhoc task summary table.
+ *
+ * @package    tool_lockstats
+ * @author     Ilya Tregubov <ilyatregubov@catalyst-au.net>
+ * @copyright  2019 Catalyst IT
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class adhoc_tasks extends html_table {
+    /**
+     * Constructor
+     */
+    public function __construct() {
+        global $DB;
+
+        parent::__construct();
+
+        $this->head = [
+            get_string('table_classname', 'tool_lockstats'),
+            get_string('component', 'tool_task'),
+            get_string('table_queuedup', 'tool_lockstats'),
+            get_string('table_processed', 'tool_lockstats'),
+            get_string('table_latency', 'tool_lockstats')
+        ];
+
+        $this->attributes['class'] = 'admintable generaltable';
+
+        $data = [];
+
+        $latencysubquery = "SELECT max(latency/lockcount)
+                              FROM {tool_lockstats_history}
+                             WHERE classname = classes.classname";
+
+        $attemptedsubquery = "SELECT COUNT(*)
+                                FROM {tool_lockstats_locks}
+                               WHERE classname = classes.classname";
+
+        $attemptedqueuedupsubquery = "SELECT COUNT(ta.*) AS queueattempted 
+                                        FROM {task_adhoc} ta
+                                        JOIN {tool_lockstats_locks} tll
+                                          ON tll.resourcekey = CONCAT('adhoc_',ta.id)
+                                       WHERE ta.classname = classes.classname";
+
+        $queuedupsubquery = "SELECT COUNT(*)
+                               FROM {task_adhoc}
+                              WHERE classname = classes.classname";
+
+        $classessubquery = "SELECT classname,
+                                   component
+                              FROM {task_adhoc}
+                          GROUP BY classname, component
+                             UNION
+                            SELECT classname,
+                                   component
+                              FROM {tool_lockstats_locks}
+                             WHERE resourcekey LIKE '%adhoc%'
```

Comment:
1) use the sql_like

2) can it be tighter so less chance of false positives:

LIKE 'adhoc_%'